llvmlibc-samples/src/llvmlibc/baremetal-semihosting/crt0llvmlibc.c

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2024, Arm Limited and affiliates.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+extern int main(int argc, char** argv);
+
+extern void _platform_init();
+
+extern char __data_source[];
+extern char __data_start[];
+extern char __data_end[];
+extern char __data_size[];
+extern char __bss_start[];
+extern char __bss_end[];
+extern char __bss_size[];
+extern char __tls_base[];
+extern char __tdata_end[];
+extern char __tls_end[];
+
+void _start(void) {
+  memcpy(__data_start, __data_source, (uintptr_t) __data_size);
```

Comment:
Pedantry: formally, memcpy and memset take a `size_t`, not a `uintptr_t`.

Reply:
Thanks will fix.